src/common/src/array/data_chunk_iter.rs

```diff
@@ -296,29 +296,21 @@ impl Row {
     {
         let mut hasher = hash_builder.build_hasher();
         for datum in &self.0 {
-            datum.hash(&mut hasher);
+            hash_datum(datum, &mut hasher);
         }
         HashCode(hasher.finish())
     }
 
     /// Compute hash value of a row on corresponding indices.
-    pub fn hash_by_indices<H>(
-        &self,
-        hash_indices: &[usize],
-        hash_builder: &H,
-    ) -> super::ArrayResult<HashCode>
+    pub fn hash_by_indices<H>(&self, hash_indices: &[usize], hash_builder: &H) -> HashCode
     where
         H: BuildHasher,
     {
         let mut hasher = hash_builder.build_hasher();
         for idx in hash_indices {
-            let datum = self.0.get(*idx);
-            match datum {
-                Some(datum) => datum.hash(&mut hasher),
-                None => bail!("index {} out of row bound", idx),
-            }
+            hash_datum(&self.0[*idx], &mut hasher);
```

Comment:
What if idx is greater than the length of the row? 🤔

Reply:
Indexing with `[..]` will panic if out of bound.

Reply:
Should we panic here? Or should we return an error?

Reply:
I think panicking here is okay since the indices are inferred or specified by some other components most of the time. If this fails, there must be a bug in our system and we're unable to recover it.